src/Factories/FunFunFactory.php

```diff
@@ -2161,4 +2170,18 @@ private function newMembershipAuthorizer(): MembershipAuthorizer {
 		);
 	}
 
+	/**
+	 * @return string[]
+	 */
+	private function getEmailAddressBlockList(): array {
+		// Backwards compatibility check for legacy config key
+		// Remove this check (and maybe inline this method) once
+		// we have changed all configuration files (also in the infrastructure repo)
+		// see https://phabricator.wikimedia.org/T352788
+		if ( isset( $this->config['email-address-blacklist'] ) ) {
```

Comment:
Should this be the other way around? Eg we check for the blocklist and if it doesn't exist fall back to the old one.